File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/access/annotationprovider/AnnotationProvider.java
package ru.tggc.telegrambotcore.access.annotationprovider;

import ru.tggc.telegrambotcore.annotation.handle.HandleMeta;

import java.lang.reflect.Method;

public interface AnnotationProvider {

    boolean supports(Method m);

    HandleMeta extractMeta(Method m);
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/access/annotationprovider/AnnotationProvider.kt
package ru.tggc.telegrambotcore.access.annotationprovider

import ru.tggc.telegrambotcore.annotation.handle.HandleMeta
import java.lang.reflect.Method

/**
 * Интерфейс для работы с аннотациями хэндлеров
 */
interface AnnotationProvider {
    fun supports(m: Method): Boolean

    fun extractMeta(m: Method): HandleMeta?
}
